sippy/SipAddress.py

```diff
@@ -163,3 +164,11 @@ def setParam(self, name, value = None):
     def delParam(self, name):
         del self.params[name]
         self.params_order.remove(name)
+
+    def getQ(self):
+        if self.q == None:
+            try:
```

Comment:
Is there any reason to hide error and fake q value if some garbage is received? None of the other SIP headers have any try/except logic and there is a good reason to not do it.

Reply:
Also, by making copy into self.q and not using it in the str() / localStr() methods is a good way to introduce a bug. Since setting x.q = 100.0 won't have any effect. I'd suggest something like this instead:

```python
                k_v = l.split('=', 1)
                if len(k_v) == 2:
                    k, v = k_v
                else:
                    k = k_v[0]
                    v = None
                if k == 'q':
                    v = float(v)
```
You would also need proper `'%.1f' % v` in the localStr().